index.js

```diff
@@ -56,8 +56,12 @@ module.exports = function (request, options) {
 
     function onEnd(err) {
       if(err) reject(err);
-      cleanup();
-      return resolve({fields, files});
+      Promise.all(filePromises)
```

Comment:
Using `Promise` with empty `filePromises` is slower than `resolve` immediately.
Smt like:
```
if (filePromises.length > 0) {
  return Promise.all(filePromises)
                        .then(files => resolve({fields, files}))
                        .catch(reject)
 } else {
  return resolve({fields, files: []})
}
```